### PR TITLE
Tooltip: Sort decimals using standard numeric compare

### LINKED ISSUE #48489: When the unit is percentage (percent(0.0-1.0)), the sorting of Tooltip does not work
**What happened**:
QQ20220429-162631

**What you expected to happen**:
values are sorted from largest to smallest
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.5.1
- Data source type & version: prometheus:2.34.0
- OS Grafana is installed on: docker
- User OS & Browser: MacOS:12.3.1    Microsoft Edge:101.0.1210.32
- Grafana plugins: Time series
- Others:

### PR DESCRIPTION
**Pull request by dprokop**
Fixes https://github.com/grafana/grafana/issues/48489 caused but the way `Intl.Collator` compares decimal numbers, as described in https://stackoverflow.com/questions/40107588/intl-collator-and-natural-sort-with-numeric-option-sorts-incorrectly-with-decima

### CODE REVIEW

File: packages/grafana-data/src/utils/arrayUtils.test.ts [hunk 1]
```diff
@@ -7,21 +7,24 @@ describe('arrayUtils', () => {
...
+      ${SortOrder.Descending} | ${testArrayString}                | ${['10', '9', '8', '7', '6', '5', '4', '3', '2', '1', ' ', null, undefined]}
+      ${SortOrder.Ascending}  | ${testArrayFloatingPoint}         | ${['0.5', '1.0', '2', '2.7', null, ' ', undefined]}
+      ${SortOrder.Descending} | ${testArrayFloatingPoint}         | ${['2.7', '2', '1.0', '0.5', null, ' ', undefined]}
+      ${SortOrder.Ascending}  | ${testArrayText}                  | ${['bar', 'baz', 'foo', null, ' ', undefined]}
+      ${SortOrder.Descending} | ${testArrayText}                  | ${['foo', 'baz', 'bar', null, ' ', undefined]}
+      ${SortOrder.Ascending}  | ${testArrayMixed}                 | ${['0.5', 1, '1.5', 'bar', 'baz', 'foo', null, ' ', undefined]}
+      ${SortOrder.Descending} | ${testArrayMixed}                 | ${['foo', 'baz', 'bar', '1.5', 1, '0.5', null, ' ', undefined]}
+      ${SortOrder.Ascending}  | ${testArrayVariableFloatingPoint} | ${[1.1234567, 1.23456, 2, 2.7, null, ' ', undefined]}
+      ${SortOrder.Descending} | ${testArrayVariableFloatingPoint} | ${[2.7, 2, 1.23456, 1.1234567, null, ' ', undefined]}
```
Comment (joshhunt) [inline packages/grafana-data/src/utils/arrayUtils.test.ts:27]: FYI, this diff is easier to understand if you "hide whitespace" in Github UI.

**Review by joshhunt — APPROVED** [review]
What a funny/annoying issue with Intl.Collator. The SO answer and Firefox bug report is very interesting, thanks for sharing!

File: packages/grafana-data/src/utils/arrayUtils.ts [hunk 2]
```diff
@@ -12,6 +12,8 @@ export function moveItemImmutably<T>(arr: T[], from: number, to: number) {
  * Null/undefined/empty string values are always sorted to the end regardless of the sort order provided
  */
 const collator = new Intl.Collator(undefined, { numeric: true, sensitivity: 'base' });
+const defaultCompare = (a: number, b: number) => (a > b ? 1 : a < b ? -1 : 0);
```
Comment (leeoniya) [inline packages/grafana-data/src/utils/arrayUtils.ts:15]: you can simply do `=> a - b`

**Comment by leeoniya** [comment]
wow, that `Intl.Collator` behavior is **wild**! :sob:

File: packages/grafana-data/src/utils/arrayUtils.ts [hunk 3]
```diff
@@ -12,6 +12,8 @@ export function moveItemImmutably<T>(arr: T[], from: number, to: number) {
  * Null/undefined/empty string values are always sorted to the end regardless of the sort order provided
  */
 const collator = new Intl.Collator(undefined, { numeric: true, sensitivity: 'base' });
+const defaultCompare = (a: number, b: number) => a - b;
```
Comment (joshhunt) [inline packages/grafana-data/src/utils/arrayUtils.ts:15]: Sorry for not spotting this the first time, but maybe calling this `numericCompare` might be more accurate? This isn't actually the default comparator.

**Review by joshhunt — APPROVED** [review]

**Pull request merged**